submissions/tests/test_api.py

```diff
@@ -68,11 +68,7 @@ def test_get_submission_and_student(self):
         retrieved = api.get_submission_and_student(submission['uuid'])
         self.assertItemsEqual(submission, retrieved)
 
-        # Should raise an exception if uuid is malformed
-        with self.assertRaises(api.SubmissionInternalError):
-            api.get_submission_and_student(u'no such uuid')
-
-        # Should raise a different exception if the student item does not exist
```

Comment:
This fix is time-sensitive, so I'm not bothering to suss out the refactoring needed to ensure all our failures are the correct type. Malformed and correctly-formed-but-not-present values both return NotFound for now.